core/src/Kokkos_CopyViews.hpp

```diff
@@ -644,211 +642,48 @@ void view_copy(const DstType& dst, const SrcType& src) {
     Kokkos::Impl::throw_runtime_exception(ss.str());
   }
 
+  using ExecutionSpace =
+      std::conditional_t<DstExecCanAccessSrc, dst_execution_space,
+                         src_execution_space>;
+
   // Figure out iteration order in case we need it
-  int64_t strides[DstType::rank + 1];
-  dst.stride(strides);
-  Kokkos::Iterate iterate;
-  if (std::is_same_v<typename DstType::array_layout, Kokkos::LayoutRight>) {
-    iterate = Kokkos::Iterate::Right;
-  } else if (std::is_same_v<typename DstType::array_layout,
-                            Kokkos::LayoutLeft>) {
-    iterate = Kokkos::Iterate::Left;
-  } else if (std::is_same_v<typename DstType::array_layout,
-                            Kokkos::LayoutStride>) {
-    if (strides[0] > strides[DstType::rank - 1])
-      iterate = Kokkos::Iterate::Right;
-    else
-      iterate = Kokkos::Iterate::Left;
-  } else {
-    if (std::is_same_v<typename DstType::execution_space::array_layout,
-                       Kokkos::LayoutRight>)
-      iterate = Kokkos::Iterate::Right;
-    else
-      iterate = Kokkos::Iterate::Left;
-  }
+  Kokkos::Iterate iterate = get_iteration_order(dst);
 
   if ((dst.span() >= size_t(std::numeric_limits<int>::max())) ||
       (src.span() >= size_t(std::numeric_limits<int>::max()))) {
-    if (DstExecCanAccessSrc) {
-      if (iterate == Kokkos::Iterate::Right)
-        Kokkos::Impl::ViewCopy<
-            typename DstType::uniform_runtime_nomemspace_type,
-            typename SrcType::uniform_runtime_const_nomemspace_type,
-            Kokkos::LayoutRight, dst_execution_space, DstType::rank, int64_t>(
-            dst, src);
-      else
-        Kokkos::Impl::ViewCopy<
-            typename DstType::uniform_runtime_nomemspace_type,
-            typename SrcType::uniform_runtime_const_nomemspace_type,
-            Kokkos::LayoutLeft, dst_execution_space, DstType::rank, int64_t>(
-            dst, src);
-    } else {
-      if (iterate == Kokkos::Iterate::Right)
-        Kokkos::Impl::ViewCopy<
-            typename DstType::uniform_runtime_nomemspace_type,
-            typename SrcType::uniform_runtime_const_nomemspace_type,
-            Kokkos::LayoutRight, src_execution_space, DstType::rank, int64_t>(
-            dst, src);
-      else
-        Kokkos::Impl::ViewCopy<
-            typename DstType::uniform_runtime_nomemspace_type,
-            typename SrcType::uniform_runtime_const_nomemspace_type,
-            Kokkos::LayoutLeft, src_execution_space, DstType::rank, int64_t>(
-            dst, src);
-    }
+    if (iterate == Kokkos::Iterate::Right)
```

Comment:
If `get_iteration_order` can be made `constexpr`, then this line could be modified to use `if constexpr`.